docs/adr/001_error_handling.md

```diff
@@ -0,0 +1,97 @@
+* ... is implemented by multiple concrete types such as `InMemoryLogExporter`, `OtlpLogExporter`, `StdOutLogExporter` that have different error requirements - for instance, an `OtlpLogExporter` will experience network failures, an `InMemoryLogExporter` will not 
+* Potentially has operations on the API that, either in the direct implementation, or in a derived utility that utilises the direct implementation, call _multiple_ API actions and therefore need to return an aggregated log type
+
+Today, we have a situation where a single error type is used per API-trait, and some methods simply swallow their errors. In the example above of `LogExporter`, `shutdown` swallows errors, and `export` returns the `LogError` type, a type that could conceptually be thought of as belonging to the entire trait, not a particular method. For the exporters, the [opentelemetry-specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/sdk.md#export) tells us that they need to indicate success or failure, with a distinction made between 'failed' and 'timed out'. 
+
+There are also similar examples in the builders for providers and exports. 
+
+## Related Work
+
+* #2564 
+* #2561 
+* #2381
+
+## Considered Options
+
+**Option 1: Continue as is**
+Continue the status quo, returning a mix of either nothing or the trait-wide error type. This is inconsistent and limits the caller's ability to handle errors. 
+
+**Option 2: Extend trait-wide error type to all methods on trait**
+In this option we keep the existing error type, add it to the remaining methods on the trait, and extend the error type to include errors covering the new error conditions. This will mean that callers will have to know how and when to discard errors from a particular API call based on an understanding of which subset of errors that particular call can make. 
+
+Conversely, it will reduce the number of error types in the code base. 
+
+**Option 3: Introduce an error-type per fallible operation, aggregate these into a single trait-wide error type**
+
+For example, in the above we'd have something like:
+```rust
+pub trait LogExporter {
+        
+	fn export(...) -> Result<..., ExportError>;
+	fn shutdown(...) -> Result<..., ShutdownError>
+}
+
+// Concrete errors for an export operation
+pub enum ExportError {
+    // The distinction between failure and timed out is part of the OTEL spec
+    // we need to meet. 
+
+    ExportFailed,  
+    
+    ExportTimedOut(Duration),
+	
+	// Allow impls to box up errors that can't be logically mapped
+	// back to one of the APIs errors 
+	#[error("Unknown error (should not occur): {source:?}")] 
+	Unknown { 
+		source: Box<dyn std::error::Error + Send + Sync>, 
+	},
+}
+
+// Aggregate error type for convenience 
+// **Note**: This will be added in response to need, not pre-emptively
```

Comment:
this maybe moved to a "Future addition" section, so it is easy to follow what the actual outcome, and curious readers/contributors can look at Future section for potential improvements to be added.

Reply:
I think this was a bit of a rabbit hole that feels unlikely with a reasonably obvious solution, so i've dropped it completely.